anomalib/models/components/base/anomaly_module.py

```diff
@@ -32,14 +35,14 @@ class AnomalyModule(pl.LightningModule, ABC):
     Acts as a base class for all the Anomaly Modules in the library.
     """
 
-    def __init__(self):
+    def __init__(self) -> None:
         super().__init__()
         logger.info("Initializing %s model.", self.__class__.__name__)
 
         self.save_hyperparameters()
         self.model: nn.Module
-        self.loss: Tensor
-        self.callbacks: List[Callback]
+        self.loss: nn.Module
```

Comment:
I don't understand this change. As far as I know, the `loss` attribute holds the value of the loss between epochs, so it should be a Tensor.

Reply:
in almost all of the implementations, we use `self.loss` to compute the loss. For example, here
https://github.com/openvinotoolkit/anomalib/blob/ce70716dbcb8fca2e2b2437eed356c0d0d9f6ca3/anomalib/models/draem/lightning_model.py#L41

When we define this as `Tensor`, pylint and mypy complains, because we use this as `Callable`

Reply:
If PL uses this under the hood, we could then rename this to `self.loss_func`